Index: src/contrib/gentoo/layman/xupnpd.xml
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/xml
Index: src/contrib/gentoo/layman/xupnpd.xml
===================================================================
--- src/contrib/gentoo/layman/xupnpd.xml	(revision 372)
+++ src/contrib/gentoo/layman/xupnpd.xml	(working copy)

Property changes on: src/contrib/gentoo/layman/xupnpd.xml
___________________________________________________________________
Added: svn:mime-type
## -0,0 +1 ##
+application/xml
\ No newline at end of property
Index: src/contrib/gentoo/portage/net-misc/xupnpd/Manifest
===================================================================
--- src/contrib/gentoo/portage/net-misc/xupnpd/Manifest	(revision 0)
+++ src/contrib/gentoo/portage/net-misc/xupnpd/Manifest	(working copy)
@@ -0,0 +1,4 @@
+AUX makefile.patch 1043 SHA256 6c04cab9c5d02e765440925bdb7a6b800cbae4e154fd9a4ec9af687034647e69 SHA512 fde1a9b4a0e6e08bda59ff310f5d40c28937fd088dc21ff0253f524d7bb93ba5de9c603d70b2893761ef6f10a7127d9243def64f8f8ed7e1a706fa7fb9f96f41 WHIRLPOOL 8c0aad1a149ec11168a67d8395384288f2828aa69b66b2fd2bcf101aaf343f4f300335daeb6e878ceca30f04e99a00b12cb478ce002512f87b5a442468aa3e3b
+AUX xupnpd.envd 72 SHA256 e0722ef30c3270c2b12f394735c3dfec387cb50d0740505f51268989f49e93cb SHA512 8ea0e0433bcc466a9b1cb39b734b32c73d57f369e811c75990c5844886c4af2f36a1b579d8dfdffaec63f9391c63ec284e544e4183b3f0eed8f74d677a307954 WHIRLPOOL b11a030c76de40c5b436f938ac635481caa4539f488705a6f8dc6bb9e0073091a80a3e497e72309bc51fe489de615e99ff8eb595b60ac7ffdcc1e1f692eda9dc
+AUX xupnpd.init 774 SHA256 6fc98db2f44ab758c1fc843f341988156c9973e066b6db417ad3ec03f2de550a SHA512 797df0401dfed343a30e12d90eb745a474a80e3c33116c514c7f7f18bc4d7463b871ffbd0ccdf7c04e07849f073bc16c1a467ea97058e2e67ddb9fd1f98605a7 WHIRLPOOL 6c090d9e4831add5118026461f02c48552fa45eecbd0911e01a5e41907316cb04ea4c38939f5e003b458c48a81142e4574486346bb275d6f896096bb4d58b111
+EBUILD xupnpd-9999.ebuild 2937 SHA256 e63cf90669510e6890c946f4bea5e5716bf392a28b91376fc19d1eb1ca944ba1 SHA512 4defaca7f06c3a53c21aa2cc9820f688ee353b4c708551fe24265f7dc0d488b6a8664cc510ebd5d091b9253ec3bb43552ceef891bc2766ffb7bf9aa469a48e2b WHIRLPOOL c5cffccf7b4e53a593fd3e0f8c7c1911c507eb34ed71ba6faebc6c4756970875471815d99719863831b1cbb01cf397dcffa65ed5ceece891c1d8d40e2fa89c36
Index: src/contrib/gentoo/portage/net-misc/xupnpd/files/makefile.patch
===================================================================
--- src/contrib/gentoo/portage/net-misc/xupnpd/files/makefile.patch	(revision 0)
+++ src/contrib/gentoo/portage/net-misc/xupnpd/files/makefile.patch	(working copy)
@@ -0,0 +1,30 @@
+Index: Makefile
+===================================================================
+diff --git a/trunk/xupnpd/src/Makefile b/trunk/xupnpd/src/Makefile
+--- a/trunk/xupnpd/src/Makefile	(revision 368)
++++ b/trunk/xupnpd/src/Makefile	(working copy)
+@@ -1,8 +1,8 @@
+-LUA     = lua-5.1.4
+-#LUA     = lua-5.2.0
++CC ?= gcc
++CXX ?= g++
++
+ STATIC  = false
+ 
+-CFLAGS  = -fno-exceptions -fno-rtti -O2 -I$(LUA) -L$(LUA)
+ SRC     = main.cpp soap.cpp mem.cpp mcast.cpp luaxlib.cpp luaxcore.cpp luajson.cpp luajson_parser.cpp
+ STAGING_DIR = /u01//home/shocker/staff/openwrt/staging_dir
+ LUAMYCFLAGS = -DLUA_USE_LINUX
+@@ -13,10 +13,8 @@
+ endif
+ 
+ x86:
+-	make -C $(LUA) CC=gcc a
+-	gcc -O2 -c -o md5.o md5c.c
+-	gcc $(CFLAGS) -DWITH_LIBUUID -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -o xupnpd $(SRC) md5.o -llua -ldl -lm -luuid
+-	strip xupnpd
++	$(CC) $(CFLAGS) -c -o md5.o md5c.c
++	$(CXX) $(CXXFLAGS) $(LDFLAGS) -fno-exceptions -fno-rtti -DWITH_LIBUUID -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -o xupnpd $(SRC) md5.o -llua -ldl -lm -luuid
+ 
+ # Cygwin
+ x86win:
Index: src/contrib/gentoo/portage/net-misc/xupnpd/files/xupnpd.envd
===================================================================
--- src/contrib/gentoo/portage/net-misc/xupnpd/files/xupnpd.envd	(revision 0)
+++ src/contrib/gentoo/portage/net-misc/xupnpd/files/xupnpd.envd	(working copy)
@@ -0,0 +1 @@
+CONFIG_PROTECT="/usr/share/xupnpd/xupnpd.lua /usr/share/xupnpd/plugins"
Index: src/contrib/gentoo/portage/net-misc/xupnpd/files/xupnpd.init
===================================================================
--- src/contrib/gentoo/portage/net-misc/xupnpd/files/xupnpd.init	(revision 0)
+++ src/contrib/gentoo/portage/net-misc/xupnpd/files/xupnpd.init	(working copy)
@@ -0,0 +1,25 @@
+#!/sbin/runscript
+# Copyright 1999-2010 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License, v2 or later
+
+runas_user=xupnpd:xupnpd
+rundir=/run/xupnpd
+pidfile=${rundir}/xupnpd.pid
+
+command="/usr/sbin/xupnpd"
+start_stop_daemon_args="--env XUPNPDROOTDIR=/usr/share/xupnpd --user ${runas_user}"
+
+description="xupnpd - eXtensible UPnP agent"
+
+depend() {
+	need localmount net
+	after bootmisc
+	use logger
+}
+
+start_pre() {
+	checkpath --directory --owner ${runas_user} --mode 0755 ${rundir}
+#	checkpath --directory --owner ${runas_user} --mode 0775 /var/lib/xupnpd/playlists
+#	checkpath --directory --owner ${runas_user} --mode 0775 /var/lib/xupnpd/playlists/example
+#	checkpath --directory --owner ${runas_user} --mode 0775 /usr/share/xupnpd/plugins
+}
Index: src/contrib/gentoo/portage/net-misc/xupnpd/xupnpd-9999.ebuild
===================================================================
--- src/contrib/gentoo/portage/net-misc/xupnpd/xupnpd-9999.ebuild	(revision 0)
+++ src/contrib/gentoo/portage/net-misc/xupnpd/xupnpd-9999.ebuild	(working copy)
@@ -0,0 +1,119 @@
+# Copyright 1999-2013 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="4"
+inherit subversion
+
+DESCRIPTION="Light DLNA Media Server"
+HOMEPAGE="http://xupnpd.org"
+SRC_URI=""
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~x86"
+
+DEPEND="=dev-lang/lua-5.1*"
+RDEPEND="${DEPEND}"
+
+ESVN_REPO_URI="http://tsdemuxer.googlecode.com/svn/trunk/xupnpd"
+
+pkg_setup() {
+	tc-export CC CXX
+}
+
+src_unpack() {
+	subversion_src_unpack
+	subversion_wc_info
+}
+
+src_prepare() {
+	cd src || die "Can't cd to src"
+	rm -rf lua-5* || die "Faild removing bundled lua"
+	epatch "${FILESDIR}/makefile.patch" || die "Failed pathing Makefile"
+	sed -i -r -e "/^cfg\.version/s/(=[ ]*'{0,1})([^']*)('{0,1})/\1\2-r${ESVN_WC_REVISION}\3/" \
+		xupnpd.lua || die "Failed sed to change version"
+	sed -i -r -e "\
+		/^cfg/s/cfg\.ssdp_interface='lo'/cfg\.ssdp_interface='eth0'/g;\
+		/^cfg/s/cfg\.ssdp_loop=1/cfg\.ssdp_loop=0/g;\
+		/^cfg/s/cfg\.debug=1/cfg\.debug=0/g;\
+		/^cfg/s/cfg\.daemon=false/cfg\.daemon=true/" \
+		xupnpd.lua || die "Failed sed to modify config defaults"
+	sed -i -e "/core\.restart/s/\.\/xupnpd/\/usr\/sbin\/xupnpd/" \
+		ui/xupnpd_ui.lua || die "Failed sed to modify binary location"
+
+	sed -i -r -e "/^cfg\.pid_file/s/\/var\/run/\/run\/xupnpd/" \
+		xupnpd.lua || die "Failed sed to modify pid location"
+}
+
+src_compile() {
+	cd src || die "Can't cd to src"
+	emake || die "emake failed"
+}
+
+pkg_preinst() {
+	enewgroup xupnpd
+	enewuser xupnpd -1 -1 /dev/null xupnpd
+}
+
+src_install() {
+	dodoc README src/ui/*.txt
+
+	dosbin src/xupnpd
+
+	insinto /usr/share/xupnpd/
+	doins src/xupnpd*.lua
+
+	dodir /var/lib/xupnpd/config
+	dodir /var/lib/xupnpd/config/postinit
+
+	dodir /usr/share/xupnpd/localmedia
+
+	insinto /usr/share/xupnpd/plugins/
+	doins src/plugins/*
+
+	insinto /usr/share/xupnpd/plugins/skel/
+	doins src/plugins/skel/*
+
+	insinto /usr/share/xupnpd/plugins/staff/
+	doins src/plugins/staff/*
+
+	insinto /usr/share/xupnpd/profiles/
+	doins src/profiles/*
+
+	insinto /usr/share/xupnpd/profiles/skel/
+	doins src/profiles/skel/*
+
+	insinto /usr/share/xupnpd/www/
+	doins src/www/*
+
+	insinto /var/lib/xupnpd/playlists/
+	doins src/playlists/*
+
+	insinto /var/lib/xupnpd/playlists/example/
+	doins src/playlists/example/*
+
+	insinto /usr/share/xupnpd/ui/
+	doins src/ui/*.lua src/ui/*.css src/ui/*.html || die "8"
+
+	dosym /usr/share/xupnpd/xupnpd.lua /etc/xupnpd/xupnpd.lua
+	dosym /var/lib/xupnpd/playlists /usr/share/xupnpd/playlists
+	dosym /var/lib/xupnpd/config /usr/share/xupnpd/config
+	dosym /var/lib/xupnpd/config /etc/xupnpd/config
+
+	newenvd "${FILESDIR}"/xupnpd.envd 99xupnpd
+	newinitd "${FILESDIR}/xupnpd.init" xupnpd
+
+	keepdir /run/xupnpd
+	fowners xupnpd:xupnpd /run/xupnpd
+
+	fowners -R root:xupnpd /usr/share/xupnpd
+	fperms 0775 /usr/share/xupnpd
+
+	fowners -R xupnpd:xupnpd /var/lib/xupnpd
+}
+
+pkg_postinst() {
+	einfo "Main configuration and plugins: /usr/share/xupnpd"
+	einfo "User-defined configuration and playlists: /var/lib/xupnpd"
+}
Index: src/contrib/gentoo/portage/profiles/repo_name
===================================================================
--- src/contrib/gentoo/portage/profiles/repo_name	(revision 0)
+++ src/contrib/gentoo/portage/profiles/repo_name	(working copy)
@@ -0,0 +1 @@
+xupnpd
